src/nodejs/unit-http/http_server.js

```diff
@@ -331,21 +335,22 @@ ServerResponse.prototype.end = function end(chunk, encoding, callback) {
         });
 
         this.finished = true;
+        this.emit('finish');
     }
 
     return this;
 };
 
 function ServerRequest(server, socket) {
     EventEmitter.call(this);
+    Readable.call(this);
```

Comment:
Readable already inherits Stream, which, in turns, inherits EventEmitter. Is it still required to inherit EventEmitter explicitly?
Currently ServerRequest overrides 'listeners' and 'resume' methods as a stubs. 'listeners' defined in EventEmitter and 'resume' - in Readable.
As soon as ServerRequest is inherited from Readable, it should have '_read' method for paused reading mode.

Reply:
@mar0x '_read' for Readable is the best way

Reply:
'_read' func can be implemented in JS: all the data already in ServerRequest._data buffer.

Reply:
@mar0x Thanks for your replay. 

For controlling executing sequence,  I have change all `process.nextTick(cb)` to `setImmediate(cb)` in `http_server.js`, it works as expected.

Also, I have compared the performance difference between `process.nextTck(cb)` and `setImmediate(cb)` in this case by `ab`, it seems no difference. Everything goes fine.

Follow your advice, I have add logger for recording event triggerd and inspecting `_data`'s in `ServerRequest.prototype.on`
![image](https://user-images.githubusercontent.com/14068989/100441041-c6243000-30e0-11eb-92e0-eaed2a9d9123.png)

The logger output after call `curl http://127.0.0.1:9003/api`
![image](https://user-images.githubusercontent.com/14068989/100441173-fe2b7300-30e0-11eb-9b54-b26a551827bb.png)

Unfortunately, the `_data`'s length is `0`. So add `_read` func on ServerRequest.prototype directly for feeding data may be not work

APPENDIX:

`udp-logger-server.js` for collecting log

```javascript
const dgram = require('dgram')
const server = dgram.createSocket('udp4')
server.bind(5678)

server.on('listening', function () {
  console.log('udp server linstening 5678.') // eslint-disable-line
})

server.on('message', function (msg, rinfo) {
  const message = msg.toString()
  console.log(`${new Date().getTime()}: ${message}`) // eslint-disable-line
})

server.on('error', err => {
  console.log('some error on udp server.', err.message) // eslint-disable-line
  server.close()
})
```

`logger.js` for send log in http_server.js

```javascript
const dgram = require('dgram')
const client = dgram.createSocket('udp4')

client.on('close', function () {
  console.log('udp client closed.') // eslint-disable-line
})

client.on('error', function () {
  console.log('some error on udp client.')  // eslint-disable-line
})

const send = message => {
  client.send(message, 0, message.length, 5678, '127.0.0.1')
}

module.exports.info = send

send('hello world from nodejs')
```

Reply:
Please do not made all the changes in this pull request. I don't see reason to blindly change `process.nextTick()` for `setImmediate()`.

Reply:
`_data.length` is `0` for GET requests because there is no request body.

Reply:
@mar0x Thanks for your reply.

I will create new PR and explaining the reason for each changes